src/Operation/Delete.php

```diff
@@ -117,18 +117,36 @@ public function execute(Server $server)
             throw UnsupportedException::collationNotSupported();
         }
 
+        $bulk = new Bulk();
+        $bulk->delete($this->filter, $this->createDeleteOptions());
+
+        $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $this->createExecuteOptions());
+
+        return new DeleteResult($writeResult);
+    }
+
+    public function getCommandDocument()
+    {
+        return ['delete' => $this->collectionName, 'deletes' => [['q' => $this->filter] + $this->createDeleteOptions()]];
```

Comment:
While I was doing a sanity check of this method against the [`delete` command reference](https://docs.mongodb.com/manual/reference/command/delete/), I noticed that it's technically missing top-level `ordered` and `writeConcern` keys.

I'm not too worried about `ordered`. We don't pass it to `BulkWrite` above, although I believe libmongoc will default it to `true` explicitly. The server default is also `true`, and it really has no effect for a single-statement write operation anyway.

I also doubt that `writeConcern` is relevant to an explain, but I do think it's worth including since it is an option on this class.

I'd suggest creating the command document as you do in the return statement and then conditionally setting `writeConcern` if it's set.